server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/JobCallerBuilder.java

```diff
@@ -36,7 +41,24 @@ public class JobCallerBuilder {
     public static JobCaller buildProcessCaller(JobContext context) {
         Map<String, String> environments = new JobEnvironmentFactory().build(context, TaskRunMode.PROCESS);
         JobUtils.encryptEnvironments(environments);
-
+        /**
+         * write JobContext to file in case of exceeding the environments size limit; set the file path in
+         * the environment instead
+         */
+        String jobContextFilePath = JobUtils.getExecutorDataPath() + "/" + StringUtils.uuid() + ".enc";
+        try {
+            FileUtils.writeStringToFile(new File(jobContextFilePath),
+                    environments.get(JobEnvKeyConstants.ODC_JOB_CONTEXT),
+                    Charset.defaultCharset());
+        } catch (Exception ex) {
+            FileUtils.deleteQuietly(new File(jobContextFilePath));
+            throw new RuntimeException("Failed to write job context to file: " + jobContextFilePath, ex);
+        }
+        environments.put(JobEnvKeyConstants.ODC_JOB_CONTEXT_FILE_PATH,
+                JobUtils.encrypt(environments.get(JobEnvKeyConstants.ENCRYPT_KEY),
+                        environments.get(JobEnvKeyConstants.ENCRYPT_SALT), jobContextFilePath));
+        // remove JobContext from environments
+        environments.remove(JobEnvKeyConstants.ODC_JOB_CONTEXT);
```

Comment:
this remove operation is wired, why not put it in env build function?

Reply:
Fixed it.